src/main/java/dev/opentrading/core/spring/CoreApplicationRunner.java

```diff
@@ -39,18 +43,36 @@ public void run(ApplicationArguments args) {
                     coreBootstrap.getTradingInstruments().thenAccept(tradingInstrument -> {
                         log.info("Filtered trading instruments");
 
-                        coreBootstrap.initWebSocket().thenAccept(initWebsocket -> {
+                        coreBootstrap.<Boolean> initWebSocket().thenApply(initialized -> {
```

Comment:
I think we don't need to make everything a completable future here like login, getMargin, getAllInstruments. If we're going to block on them all anyway, this chain of futures just adds boilerplate and a bunch of extra nesting. We should maybe only wrap things in futures if we want to launch tasks in parallel or need other specific capabilities.